### DISCUSSION TITLE
x-for inside vue instance

### DISCUSSION THREAD

**Discussion by swist666** [alpinejs/alpine#1363] [4. General]
Is it possible to do x-for inside a vue application?

Here is the code to reproduce the issue:

```html
<div id="app">

    <div x-data="{array: ['A','B','C']}">
        <template x-for="(item, index) in array">
            <div x-text="item"></div>
        </template>
    </div>

</div>

<script src="https://cdn.jsdelivr.net/gh/alpinejs/alpine@v2.8.2/dist/alpine.min.js" defer></script>
<script src="https://cdn.jsdelivr.net/npm/vue@2.6.12/dist/vue.js"></script>
<script>
const app = new Vue({
    el: '#app'
});
</script>
```

I get `Alpine Error: "ReferenceError: item is not defined"` with this code, when I delete Vue it works correclty.

**Comment by KevinBatdorf** [comment]
Vue will compute the `template` tags as that's a part of how Vue works. I tried adding `v-pre` to ignore it, but seems to be a bug in Vue as it wasn't ignored. It's probably worth opening an issue on the Vue repo

**Reply by KevinBatdorf** [reply]
This looks related: https://github.com/vuejs/vue/issues/9427

**Reply by swist666** [reply]
Yes v-pre does not work :(

**Reply by daronspence** [reply]
@swist666 did you find a workaround for this? I have a huge legacy app that I would like to swap out some stuff for alpine but the entire app is wrapped as a vue root element. 

Alpine works fine for everything except loops since Vue rips out the template tags :(

**Reply by swist666** [reply]
nope :( everything seems to work fine except the x-for loop
